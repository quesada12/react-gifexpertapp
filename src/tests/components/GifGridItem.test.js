import { shallow } from "enzyme"
import {GifGridItem} from '../../components/GifGridItem'


describe('Pruebas en el Componente GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    
    test('Debe mostrarse el componente de la forma correcta', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe de tener un <p> con el Title', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    })

    test('Debe de tener una <img /> igual al url y al alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('Debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toContain('animate__fadeIn');
    })
    
    

})
