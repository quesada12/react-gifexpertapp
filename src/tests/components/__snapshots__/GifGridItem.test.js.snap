// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente GifGridItem Debe mostrarse el componente de la forma correcta 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
